acme/api/handler_test.go

```diff
@@ -760,13 +798,21 @@ func TestHandlerGetChallenge(t *testing.T) {
 				assert.Equals(t, ae.Identifier, prob.Identifier)
 				assert.Equals(t, ae.Subproblems, prob.Subproblems)
 				assert.Equals(t, res.Header["Content-Type"], []string{"application/problem+json"})
-			} else {
+			} else if res.StatusCode >= 200 {
 				expB, err := json.Marshal(tc.ch)
 				assert.FatalError(t, err)
 				assert.Equals(t, bytes.TrimSpace(body), expB)
-				assert.Equals(t, res.Header["Link"], []string{fmt.Sprintf("<https://ca.smallstep.com/acme/authz/%s>;rel=\"up\"", tc.ch.AuthzID)})
-				assert.Equals(t, res.Header["Location"], []string{url})
 				assert.Equals(t, res.Header["Content-Type"], []string{"application/json"})
+				switch tc.ch.Status {
+				case "processing":
```

Comment:
I might be misunderstanding the code, but if the status is `invalid` we still return a 200, right? So shouldn't we have another case here? Not really sure what you'd verify -- maybe that those fields aren't set? But just having the case with a "do nothing" will help to clarify.

Reply:
Yes, I think I need to add the case for "invalid" here. Good catch. These tests pass though, so I don't think we're actually testing that path, currently.

Reply:
But that would happen below. If the challenge is processing up here we return the challenge in the processing state with retryafter information.

Reply:
yeah, I was just saying 1) it would be nice to have a case for invalid, and 2) it would be nice to have an invalid test.